scrapper.py

```diff
@@ -25,39 +36,175 @@ class Crawler:
     """
     Crawler implementation
     """
+
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        """
+        Extract URL
+        """
+        sections_bs = article_bs.find('div', {'class': 'sections'})
+        urls_bs = sections_bs.find_all('a')
+        for url_bs in urls_bs:
+            if len(self.urls) < self.max_articles:
+                if url_bs['href'] not in self.urls:
+                    self.urls.append(url_bs['href'])
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+
+        for seed_url in self.seed_urls:
+            response = requests.get(seed_url, headers=HEADERS)
+            if not response.ok:
+                print("Failed")
+
+            article_bs = BeautifulSoup(response.text, 'lxml')
+            self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    """
+    HTMLWithPDFParser implementation
+    """
+
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(self.article_url, self.article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        """
+        Extract text
+        """
+        body_bs = article_bs.find('body')
+        links_bs = body_bs.find_all('a')
+        print(links_bs)
+        # print(article_bs)
+        # galley = article_bs.find('div', {'class': 'item galleys'})
+        # print(galley)
+        pdf_link = ''
+        try:
+            for link_bs in links_bs:
+                try:
+                    # if 'obj_galley_link pdf' in link_bs['href']:
+                    find_link = re.findall(r'article/view/\d+/\d+', link_bs['href'])
+                    if find_link:
+                        pdf_link = link_bs['href']
+                except KeyError:
+                    print("Key Error")
+                    continue
+            if not pdf_link:
+                print('Error')
+            response_pdf = requests.get(pdf_link, HEADERS)
+            # print(response_pdf)
+            # print("ЗДЕСЬ КОНЕЦ СТАТЬИИИИИИИИИИИИИИИИИИИИИ")
+            pdf_bs = BeautifulSoup(response_pdf.text, 'lxml')
+
+            download_pdf = pdf_bs.find('a', {'class': 'download'})['href']
+
+            pdf = PDFRawFile(download_pdf, self.article_id)
```

Comment:
to this moment, I have run you implementation: it produces text only for a half of URLs, other files are just empty